Client/Frontend/Browser/Punycode.swift

```diff
@@ -0,0 +1,197 @@
+    }
+
+    func encode(input: String) -> String {
+        var output = ""
+        var d : Int = 0
+        var extendedChars = [Int]()
+        for c in input.unicodeScalars {
+            if Int(c.value) < initialN {
+                d += 1
+                output.append(c)
+            } else {
+                extendedChars.append(Int(c.value))
+            }
+        }
+        if extendedChars.count == 0 {
+            return output
+        }
+        if d > 0 {
+            output.append(delimiter)
+        }
+
+        var n = initialN
+        var delta = 0
+        var bias = initialBias
+        var h : Int = 0
+        var b : Int = 0
+
+        if d > 0 {
+            h = output.unicodeScalars.count - 1
+            b = output.unicodeScalars.count - 1
+        } else {
+            h = output.unicodeScalars.count
+            b = output.unicodeScalars.count
+        }
+
+        for _ = 0; h < input.unicodeScalars.count; {
+            var char = Int(0x7fffffff)
+            for c in input.unicodeScalars {
+                let ci = Int(c.value)
+                if char > ci && ci >= n {
+                    char = ci
+                }
+            }
+            delta = delta + (char - n) * (h + 1)
+            if delta < 0 {
+                print("error: invalid char:")
+                output = ""
+                return output
+            }
+            n = char
+            for c in input.unicodeScalars {
+                let ci = Int(c.value)
+                if ci < n || ci < initialN {
+                    delta += 1
+                    continue
+                }
+                if ci > n {
+                    continue
+                }
+                var q = delta
+                for var k = base; ; k += base {
+                    let t = max(min(k - bias, tMax), tMin)
+                    if q < t {
+                        break
+                    }
+                    let code = t + ((q - t) % (base - t))
+                    output.append(toValue(code))
+                    q = (q - t) / (base - t)
+                }
+                output.append(toValue(q))
+
+                bias = self.adapt(delta, numPoints: h + 1, firstTime: h == b)
+                delta = 0
+                h += 1
+            }
+            delta += 1
+            n += 1
+        }
+        return output
+    }
+
+    func decode(punycode: String) -> String {
+        var input = [Character](punycode.characters)
+        var output = [Character]()
+        var i = 0
+        var n = initialN
+        var bias = initialBias
+        var pos = 0
+        if let ipos = input.indexOf(self.delimiter) {
+            pos = ipos
+            output.appendContentsOf(input[0 ..< pos++])
+        }
+        var outputLength = output.count
+        let inputLength = input.count
+        while pos < inputLength {
+            let oldi = i
+            for var w = 1, k = base; ; k += base {
```

Comment:
Swift 3 removes C-style `for`-loops altogether. You should rewrite this to be more idiomatic.